src/services/workspace/WorkspaceService.js

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) Cosmo Tech.
+// Licensed under the MIT license.
+
+
+import DatasetService from '../dataset/DatasetService';
+import {STORAGE_ROOT_DIR_PLACEHOLDER, UPLOAD_FILE_STATUS_KEY, UploadFileUtils, WorkspaceFileUtils} from "../../utils";
+
+class WorkspaceService {
+  constructor (apiService) {
+    this.apiService = apiService;
+    this.workspaceApi = new apiService.WorkspaceApi();
+    this.datasetApi = new apiService.DatasetApi();
+  }
+
+  async uploadWorkspaceFile (organizationId, workspaceId, file, overwrite, destination) {
+    const workspace = await this.workspaceApi.uploadWorkspaceFile(organizationId, workspaceId, file, { overwrite: overwrite, destination: destination });
+    return workspace;
+  }
+
+  async updateDatasetPartFile (organisationId, dataset, datasetFile, setDatasetFile,
+    datasetId, setDatasetId, parameterId, connectorId, scenarioId, workspaceId) {
+    if (datasetFile.status === UPLOAD_FILE_STATUS_KEY.READY_TO_UPLOAD) {
+      await this.updateFileWithUpload(organisationId, datasetFile, setDatasetFile, dataset, datasetId,
+        parameterId, connectorId, scenarioId, workspaceId, datasetFile.name);
+    } else if (datasetFile.status === UPLOAD_FILE_STATUS_KEY.READY_TO_DELETE) {
+      await UploadFileUtils.updateFileWithDelete(datasetFile, setDatasetFile, dataset, datasetId,
+        setDatasetId);
+    }
+  }
```

Comment:
As we previously discussed, maybe this king of function is too dependent of the structure of the ScenarioParameters state. This simple if/else logic could probably be left in the webapp